apiServer.js

```diff
@@ -46,6 +46,19 @@ function humanOutput(res, json) {
 }
 
 var packages = JSON.parse(fs.readFileSync('public/packages.min.json', 'utf8')).packages;
+// append SRI data to each pacakge
+_.each(packages, function(library) {
```

Comment:
We'll need to append the SRI data on the fly to prevent both CPU and IO blocking

Reply:
I created a function to load SRI data and then call it on the fly, as late as possible, inside each `app.get()` call and the performance was much worse on my local workstation. I thought about it and it made sense, since now it was making a bunch of synchronous file reads _for every request_. I recommend we append the SRI data during server startup and keep it memory along with the rest of the library data, as this change does. That way the blocking only happens briefly when the API server starts or restarts. I suppose the question is, how much memory does all the SRI data add?

Reply:
I checked memory usage with `process.memoryUsage()` before loading all the SRI data and after.

memory    | before SRI | after SRI | diff
--------- | ---------- | --------- | ----
rss       |  412065792 | 539426816 | 127361024 (122MB)
heapTotal |  381935616 | 511979520 | 130043904 (124MB)
heapUsed  |  366376832 | 480901200 | 114524368 (109MB)
external  |      24764 |   2335061 |   2310297 (  2MB)

Looks like it will require an additional 109MB of memory (heapUsed), which makes sense since all the files in the sri folder add up to 88MB. Can the production servers handle this?

Reply:
By the way, if I trigger garbage collection before each call to `memoryUsage()` then overall memory usage goes down, but the difference is roughly the same.

memory | before SRI | after SRI | diff
------ | ---------- | --------- | ----
heapused | 175755600 | 275839776| 100084176 (96MB)

Reply:
As we'll cache the result on CDN page, the latency may not hurt so much, however, we don't have enough memory on heroku now :(